Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_2.h

```diff
@@ -0,0 +1,687 @@
+          it++ = face;
+          for(int i=0; i<3; ++i)
+          {
+            if(visited.find(face->neighbor(i)) == visited.end())
+            {
+              to_visit.insert(face->neighbor(i));
+              trans.insert(FTPair(face->neighbor(i), tr*neighbor_translation(face,i)));
+            }
+          }
+        }
+      } // end of while
+    } // end if(start != Face_handle())
+  } // end of function
+
+  int try_to_remove_dummy_vertices()
+  {
+    // count of dummy points in the triangulation
+    int cnt = 0;
+    for(std::size_t i=0; i<dummy_points.size(); ++i)
+    {
+      if(dummy_points[i].is_inserted())
+      {
+        if(remove(dummy_points[i].vertex()))
+          dummy_points[i].set_inserted(false);
+        else
+          ++cnt;
+      }
+    }
+
+    return cnt;
+  }
+
+  Point get_dummy_point(int i) const
+  {
+    CGAL_triangulation_precondition(0 <= i && i <= static_cast<int>(dummy_points.size()));
+    return dummy_points[i]();
+  }
+
+  bool remove(Vertex_handle v);
+
+  bool is_dummy_vertex(Vertex_handle vh) const
+  {
+    for(std::size_t i=0; i<dummy_points.size(); ++i)
+    {
+      if(dummy_points[i]() == vh->point())
+        return true;
+    }
+
+    return false;
+  }
+
+  int number_of_dummy_points() const
+  {
+    int cnt = 0;
+    for(std::size_t i=0; i<dummy_points.size(); ++i)
+    {
+      if(dummy_points[i].is_inserted())
+        ++cnt;
+    }
+
+    return cnt;
+  }
+
+protected:
+
+  Bounded_side _side_of_circle(const Face_handle f, const Point& q,
+                               const Hyperbolic_translation& translation) const
+  {
+    Point p[] = {f->vertex(0)->point(),
+                 f->vertex(1)->point(),
+                 f->vertex(2)->point()};
+
+    Hyperbolic_translation o[]= {translation * f->translation(0),
+                                 translation * f->translation(1),
+                                 translation * f->translation(2)};
+
+#if defined PROFILING_MODE
+    if(o[0].is_identity() && o[1].is_identity() && o[2].is_identity())
+      calls_predicate_identity++;
+    else
+      calls_predicate_non_identity++;
+#endif
+
+#if defined PROFILING_MODE
+    CGAL::Timer tmr;
+    tmr.start();
+#endif
+
+    Oriented_side os = this->side_of_oriented_circle(p[0], p[1], p[2], q,
+                                                     o[0], o[1], o[2], Hyperbolic_translation());
+#if defined PROFILING_MODE
+    tmr.stop();
+    if(o[0].is_identity() && o[1].is_identity() && o[2].is_identity())
+      time_predicate_identity += tmr.time();
+    else
+      time_predicate_non_identity += tmr.time();
+#endif
+
+    if(os == ON_NEGATIVE_SIDE)
+      return ON_UNBOUNDED_SIDE;
+    else if(os == ON_POSITIVE_SIDE)
+      return ON_BOUNDED_SIDE;
+    else
+      return ON_BOUNDARY;
+  }
+
+public:
+  Point dual(const Face_handle f, const Hyperbolic_translation& nboff = Hyperbolic_translation()) const
+  {
+    Point res = Construct_hyperbolic_circumcenter_2()(f->vertex(0)->point(),
```

Comment:
Need to call the traits' (`geom_traits().construct...`) + same at other places

Reply:
Checked and done.